courselab/sucury
# feat: Enhance snake's movement responsiveness

This pull request aims to address the issue #96  where the snake doesn't move as expected when keys are pressed too quickly. By introducing a movement queue, we can ensure that the latest direction input by the player is always processed, leading to a more responsive gameplay experience.

### Proposed Changes:
1. Introduced a `movement_queue` to keep track of movement directions.
2. Updated the `Snake` class' `update` method to process the latest direction from the movement queue.
3. Modified key press handling in the main loop to update the `movement_queue` rather than directly setting the snake's direction.
4. Cleaned up some extraneous whitespace and comments for clarity.

### How to Test:
1. Start the game.
2. Rapidly press different direction keys.
3. Observe that the snake now responds to the most recent direction key pressed, even when multiple keys are pressed in quick succession.
4. Ensure that the snake still cannot move in the opposite direction of its current movement (e.g., if moving right, pressing left should have no effect).
